### PR TITLE
[NG Frontend] Frontend exceptions

### PR DESCRIPTION
**Pull request by nosovmik**
### Details:
 - Introduce classes and macros for nGraph FrontEndManager exceptions.
 - Those will be exposed to python so that MO could do proper exception handing

### Tickets:
 - 52255

### CODE REVIEW

File: ngraph/frontend/frontend_manager/src/frontend_manager.cpp [hunk 1]
```diff
@@ -32,7 +25,8 @@ class FrontEndManager::Impl
 
     FrontEnd::Ptr loadByFramework(const std::string& framework, FrontEndCapFlags fec)
     {
-        FRONT_END_ASSERT(m_factories.count(framework))
```
Comment (ilyachur) [inline ngraph/frontend/frontend_manager/src/frontend_manager.cpp:35]: Can we add new tests for all new macros?

BTW, why cannot we use nGraph macros? If I remember right, ONNX importer uses nGraph macros.

Comment (nosovmik) [reply ngraph/frontend/frontend_manager/src/frontend_manager.cpp:35]: @itikhono Can you please handle this?

Comment (itikhono) [reply ngraph/frontend/frontend_manager/src/frontend_manager.cpp:35]: We have created a new execution hierarchy for frontends based on the ngraph classes to exactly detect the types of problems that arise and where (frontend/ngraph), the macros are also based on ngraph helpers.
I have added unit tests for part of macros, @nosovmik can we add this file to PR ngraph/test/frontend/paddlepaddle/exceptions.cpp?

Comment (itikhono) [reply ngraph/frontend/frontend_manager/src/frontend_manager.cpp:35]: done

File: ngraph/test/frontend/frontend_manager.cpp [hunk 2]
```diff
@@ -161,3 +162,113 @@ TEST(FrontEndManagerTest, testDefaultPlace)
...
+    } catch (...) {
+        FAIL() << "Not expected exception type.";
+    }
+}
+
+TEST(FrontEndExceptionTest, frontend_initialization_error_throw_info)
+{
+    std::string msg("msg example");
+    try {
+        EXPECT_NO_THROW(FRONT_END_INITIALIZATION_CHECK(true, msg));
```
Comment (ilyachur) [inline ngraph/test/frontend/frontend_manager.cpp:267]: Why do we have `EXPECT_NO_THROW` inside `try...catch` block?

Comment (itikhono) [reply ngraph/test/frontend/frontend_manager.cpp]: fixed

File: ngraph/frontend/frontend_manager/include/frontend_manager/frontend_exceptions.hpp [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                           const std::string& context,
+                           const std::string& explanation)
+                : CheckFailure(check_loc_info,
+                               "FrontEnd API failed with GeneralFailure: " + context,
+                               explanation)
+            {
+            }
+        };
+
+        class InitializationFailure : public CheckFailure
```
Comment (ilyachur) [inline ngraph/frontend/frontend_manager/include/frontend_manager/frontend_exceptions.hpp:28]: I think we need to export these classes.

Comment (itikhono) [reply ngraph/frontend/frontend_manager/include/frontend_manager/frontend_exceptions.hpp]: done

**Review by ilyachur — APPROVED** [review]

**Pull request merged**